pandas/core/indexes/api.py

```diff
@@ -135,13 +135,20 @@ def _get_combined_index(
     indexes = _get_distinct_objs(indexes)
     if len(indexes) == 0:
         index = Index([])
-    elif len(indexes) == 1:
+    elif len(indexes) == 1 or all_indexes_same(indexes):
```

Comment:
add some comments here as it is non-obvious what is happening

Reply:
How about this?

```suggestion
    # if unique by id or unique by value
    elif len(indexes) == 1 or all_indexes_same(indexes):
```